contributor-docs/adrs/adr-xxx-behavior-isolation.md

```diff
@@ -0,0 +1,137 @@
+
+Primer React's [recent addition of the Relative Time component](https://github.com/primer/react/pull/2484) (itself a wrapper of the [`relative-time`](https://github.com/github/relative-time-element) custom element) demonstrates that a custom element can be loaded into a server side runtime such as NodeJS without causing error, by adding [2 additional lines of code to the element itself](https://github.com/github/relative-time-element/blob/4ab3f2807a02dfd687b2b7fa04553300ff87ddbe/src/relative-time-element.ts#L2-L3):
+
+```typescript
+const root = (typeof globalThis !== 'undefined' ? globalThis : window) as typeof window
+const HTMLElement = root.HTMLElement || (null as unknown as typeof window['HTMLElement'])
+```
+
+These lines could also be added to the global context, to avoid adding it to each element.
+
+The `RelativeTime` react component uses `@lit-labs/react`, which makes integrating the component into React painless, and requires [1 line of code for integration](https://github.com/primer/react/blob/d995bb8485fa1dfd0a3637b4851c5ac539b8416d/src/RelativeTime/RelativeTime.tsx#L8):
+
+```typescript
+const RelativeTime = styled(createComponent(React, 'relative-time', RelativeTimeElement))(sx)
+```
+
+There are no additional steps required for consumers of Primer React, and the component behaves as if it is a first-class React component, supporting the same properties and events as any regular React component. The server will render the Custom Element tag (in the case of the `RelativeTime` component, `<relative-time>`) and the client is able to assign class properties and use callbacks for events, as expected.
+
+While React 18 and below require a small library like `@lit-labs/react`, due to the [lack of support for Custom Element](https://custom-elements-everywhere.com/libraries/react/results/results.html), React 19 [will likely need no such library](https://custom-elements-everywhere.com/libraries/react-experimental/results/results.html).
+
+#### Styling with `styled-components`
+
+Primer React currently uses a CSS-in-JS library called [`styled-components`](https://styled-components.com/). This library accumulates objects that map to CSS properties, and during runtime it will create a hash digest to use as the class name for the component, and inject the mapped CSS into a stylesheet on the page. This way, CSS lives in JavaScript object notation within the same file as the React Component itself. In addition to this, Primer React also uses [`styled-system`](https://styled-system.com/) which allows the use of style properties directly within a React Components props. Primer React uses the `sx` prop to allow for customisation of styles.
+
+ADR 002 describes complexity around styling Custom Elements with the styled-components library. It claimed that a custom element must use another wrapping element to apply styles.
+
+Today, Primer React's RelativeTime component can be styled using the `sx` prop, just like any other Primer React component.
```

Comment:
If we start to use custom elements for behaviors which do not have a visual components (eg everything in [primer/behaviors](https://github.com/primer/behaviors/tree/main/src) today), we be adding a new Element into the DOM to wrap the targeted area. This means the wrapping custom element will become part of the flow of the document and have styles of it's own, even though it is only intended to provide _behavior_, not _visuals_. This can become a practical issues if the element being wrapped is part of a layout with flex/grid styles, in which case the wrapper needs to be styled in such a way as to take up the same amount of space as the component being wrapped would have. Is there some way to tell the browser to completely ignore an element from the styling flow?